### PR TITLE
Fix NavMenu keyboard navigation (Fixes #479)

### LINKED ISSUE #479: fix: navigation problem in nav-menu
1. Run the examples proejct and go to the NavMenu demo
2. Tab to the expand/collapse button on a FluentNavmenu, then press cursor down to navigate the menu items.
3. On some it works, on others it does not.

### PR DESCRIPTION
**Pull request by mrpmorris**
Changed the navigator in the NavMenu so it is a FluentMenuItem instead of a button.

This allows the user to tab into the nav menu and then explore the options using the keyboard (cursor keys).

### CODE REVIEW

File: Microsoft.Fast.sln
Comment (vnbaaij) [inline Microsoft.Fast.sln:1]: You've added a new project locally (DSNavMenuTestApp) which we do not need on our side

File: examples/FluentUI.Demo.Shared/Pages/NavMenu/Examples/NavMenuDefault.razor [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 @inject ILogger<NavMenuDefault> logger;
 
-<h2>Navigation Examples</h2>
+<h2>Navigation Examples .</h2>
```
Comment (vnbaaij) [inline examples/FluentUI.Demo.Shared/Pages/NavMenu/Examples/NavMenuDefault.razor:5]: Why the space?

File: examples/FluentUI.Demo.Shared/wwwroot/sources/NavMenuDefault.razor.txt [hunk 2]
```diff
@@ -2,11 +2,11 @@
 
 @inject ILogger<NavMenuDefault> logger;
 
-<h2>Navigation Examples</h2>
+<h2>Navigation Examples .</h2>
```
Comment (vnbaaij) [inline examples/FluentUI.Demo.Shared/wwwroot/sources/NavMenuDefault.razor.txt:5]: Why the space?

File: src/Microsoft.Fast.Components.FluentUI/Components/NavMenu/FluentNavMenu.razor.cs [hunk 3]
```diff
@@ -104,8 +105,13 @@ internal async Task CollapsibleClickAsync()
         }
     }
 
-    internal void HandleSelectedChange(FluentTreeItem treeItem)
+    internal async Task HandleSelectedChange(FluentTreeItem treeItem)
```
Comment (vnbaaij) [inline src/Microsoft.Fast.Components.FluentUI/Components/NavMenu/FluentNavMenu.razor.cs:108]: Why do you want to make it async? If you do make it async, reflect that with an Async suffix for the method name

**Review by vnbaaij — CHANGES_REQUESTED** [review]

File: Microsoft.Fast.sln
Comment (mrpmorris) [reply Microsoft.Fast.sln:1]: Ah poo

File: examples/FluentUI.Demo.Shared/Pages/NavMenu/Examples/NavMenuDefault.razor (hunk 1 above)
Comment (mrpmorris) [reply examples/FluentUI.Demo.Shared/Pages/NavMenu/Examples/NavMenuDefault.razor]: Error, fixed

**Pull request merged**